test/namespace_assertion.go

```diff
@@ -65,11 +68,22 @@ func (a *NamespaceAssertion) HasNoLabel(key string) *NamespaceAssertion {
 	return a
 }
 
-func (a *NamespaceAssertion) HasResource(name string, obj runtime.Object) *NamespaceAssertion {
+func InnerResourceHasLabels(key, value string) ResourceCriterion {
+	return func(obj runtime.Object) bool {
+		metaObj, _ := meta.Accessor(obj)
+		labels := metaObj.GetLabels()
+		return labels[key] == value
+	}
+}
+func (a *NamespaceAssertion) HasResource(name string, obj runtime.Object, criteria ...ResourceCriterion) *NamespaceAssertion {
```

Comment:
The label is expected to be present at all inner resources - I guess that we don't need to have optional criteria for that and we can make it as a mandatary check for all resources. 
In other words, we can drop the parameter `criteria` and move the check inside of this `HasResource` method. With that, it won't happen that you could miss to heck it in some cases - eg here: https://github.com/codeready-toolchain/member-operator/blob/f0052f4968415f8b942659b034b5e28018343c99/pkg/controller/nstemplateset/nstemplateset_controller_test.go#L480